test/e2e/util.go

```diff
@@ -174,3 +190,41 @@ func getDeployment(t *testing.T, client client.Client, name types.NamespacedName
 	}
 	return &dep, nil
 }
+
+// Send HTTPS Request to a route and check what is returned in the Response
+func getHttpResponse(t *testing.T, route *routev1.Route) (http.Header, int, string, error) {
+	t.Helper()
+	if len(route.Spec.Host) == 0 {
+		return nil, 0, "", fmt.Errorf("route.Spec.Host is empty, cannot test route")
+	}
+	request, err := http.NewRequest("GET", "https://"+route.Spec.Host, nil)
+	if err != nil {
+		return nil, 0, "", fmt.Errorf("error creating HTTPS request %v: %v", request, err)
+	}
+
+	// Send the HTTPS request
+	timeout, _ := time.ParseDuration("10s")
+	client := &http.Client{
+		Timeout: timeout,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	// Get the response
+	response, err := client.Do(request)
```

Comment:
We can just make this `client.Get("https://"+route.Spec.Host")` dispensing with the call to `http.NewRequest()` above. The `Get()` will construct the request for you.

Reply:
I made all the other changes you requested.  Can you live with this one?  I don't think it harms.

Reply:
I think the best thing would be to remove the function. It's not that generic, and it could be a lot shorter if we just did the minimum necessary in the test. (Apologies for the duplicated comment; deleted the other one.)

Reply:
will do

Reply:
I haven't tested this, but this feels simpler:

```go
	client := &http.Client{
		Timeout: 10 * time.Second,
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
		},
	}

	// Wait for route service to respond, and when it does, check for the header that should be there
	if err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
		resp, err := client.Get(validRoute.Spec.Host)
		if err != nil {
			t.Logf("GET failed: %v, retrying...", err)
			return false, nil
		}
		defer resp.Body.Close()
		if resp.StatusCode != http.StatusOK {
			return false, fmt.Errorf("expected http.statusCode %v, got %v", http.StatusOK, resp.StatusCode)
		}
		if header := resp.Header.Get("strict-transport-security"); header != exampleHeader {
			t.Fatalf("request to %s got wrong HSTS header, should be %s, not %s", validRoute.Spec.Host, exampleHeader, header)
		}
		return true, nil
	}); err != nil {
		t.Fatalf("failed to get HSTS header: %v", err)
	}
```

(Sync tomorrow.)

Reply:
done